File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.py
def solution(A, B):
    answer = 0
    A = list(A)
    
    for _ in range(len(A)):
        if ''.join(A) == B:
            return answer 
    
        last = A.pop()
        A.insert(0, last)

        answer += 1
    return -1

File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.java
class Solution {
    public int solution(String A, String B) {
        
        if (A.equals(B)) {
            return 0;
        }
        
        int l = A.length();
        
        for (int i=1; i<l; i++) {
            A = A.substring(l-1) + A.substring(0, l-1);
            if (A.equals(B)) {
                return i;
            }
        }
        return -1;
    }
}